cmd/promtool/main.go

```diff
@@ -434,12 +446,14 @@ func checkConfig(agentMode bool, filename string) ([]string, error) {
 	return ruleFiles, nil
 }
 
-func checkTLSConfig(tlsConfig config_util.TLSConfig) error {
-	if err := checkFileExists(tlsConfig.CertFile); err != nil {
-		return errors.Wrapf(err, "error checking client cert file %q", tlsConfig.CertFile)
-	}
-	if err := checkFileExists(tlsConfig.KeyFile); err != nil {
-		return errors.Wrapf(err, "error checking client key file %q", tlsConfig.KeyFile)
+func checkTLSConfig(tlsConfig config_util.TLSConfig, checkSyntaxOnly bool) error {
+	if !checkSyntaxOnly {
+		if err := checkFileExists(tlsConfig.CertFile); err != nil {
+			return errors.Wrapf(err, "error checking client cert file %q", tlsConfig.CertFile)
+		}
+		if err := checkFileExists(tlsConfig.KeyFile); err != nil {
+			return errors.Wrapf(err, "error checking client key file %q", tlsConfig.KeyFile)
+		}
 	}
```

Comment:
to get rid of this extra nesting and also not have the `!checkSyntaxOnly` part, we could move these `checkFileExists`  below the `if len(tlsConfig.CertFile) > 0 && len(tlsConfig.KeyFile)...` checks and add after
```
if checkSyntaxOnly { break }
``` 
like you do elsewhere.
This change isn't necessary, just reduces the mental load a little bit when reading the code.